packages/@cdktf/cli-core/src/lib/models/terraform-cli.ts

```diff
@@ -106,28 +107,51 @@ export class TerraformCli implements Terraform {
         createTerraformLogHandler(phase, filter)(stderr.toString(), true);
   }
 
-  public async init(needsUpgrade: boolean, noColor?: boolean): Promise<void> {
+  public async init(
+    needsUpgrade: boolean,
+    noColor: boolean,
+    migrateState: boolean
```

Comment:
nit: We're slowly approaching a count of args here, where I'd prefer a config object over a list of args 😅